src/main/java/de/tum/in/www1/artemis/service/connectors/pyris/PyrisWebhookService.java

```diff
@@ -0,0 +1,172 @@
+        int lectureId = attachmentUnit.getLecture().hashCode();
+        int courseId = attachmentUnit.getLecture().getCourse().hashCode();
+        return new PyrisLectureUnitWebhookDTO(false, artemisBaseUrl, "", lectureUnitId, "", lectureId, "", courseId, "", "");
+    }
+
+    /**
+     * send the updated / created attachment to Pyris for ingestion if autoLecturesUpdate is enabled
+     *
+     * @param courseId               Id of the course where the attachment is added
+     * @param newAttachmentUnits     the new attachment Units to be sent to pyris for ingestion
+     * @param irisSettingsRepository the repository to get the course settings from
+     * @return true if the units were sent to pyris
+     */
+    public boolean autoUpdateAttachmentUnitsInPyris(IrisSettingsRepository irisSettingsRepository, Long courseId, List<AttachmentUnit> newAttachmentUnits) {
+        try {
+            IrisCourseSettings courseSettings = irisSettingsRepository.findCourseSettings(courseId).orElseThrow(() -> new NoSuchElementException("Course settings not found"));
+            if (courseSettings.getIrisLectureIngestionSettings() != null && courseSettings.getIrisLectureIngestionSettings().isEnabled()
+                    && Boolean.TRUE.equals(courseSettings.getIrisLectureIngestionSettings().getAutoIngestOnLectureAttachmentUpload())) {
+                return addLectureToPyrisDB(newAttachmentUnits) != null;
+            }
+        }
+        catch (NoSuchElementException e) {
+            log.debug("Course settings not found");
+        }
+        return false;
+    }
+
+    /**
+     * delete the lectures from the vector database on pyris
+     *
+     * @param attachmentUnits The attachmentUnit that got Updated / erased
+     * @return jobToken if the job was created
+     */
+    public String deleteLectureFromPyrisDB(List<AttachmentUnit> attachmentUnits) {
+        try {
+            List<PyrisLectureUnitWebhookDTO> toUpdateAttachmentUnits = new ArrayList<>();
+            attachmentUnits.stream().filter(unit -> unit.getAttachment().getAttachmentType() == AttachmentType.FILE && unit.getAttachment().getLink().endsWith(".pdf"))
+                    .forEach(unit -> {
+                        toUpdateAttachmentUnits.add(processAttachmentForDeletion(unit));
+                    });
+            if (!toUpdateAttachmentUnits.isEmpty()) {
+                return executeLectureWebhook(toUpdateAttachmentUnits);
+            }
+        }
+        catch (Exception e) {
+            log.error(e.getMessage());
+        }
+        return null;
+    }
+
+    /**
+     * adds the lectures to the vector database on pyris
+     *
+     * @param attachmentUnits The attachmentUnit that got Updated / erased
+     * @return jobToken if the job was created
+     */
+    public String addLectureToPyrisDB(List<AttachmentUnit> attachmentUnits) {
+        try {
+            if (lectureIngestionEnabled(attachmentUnits.getFirst().getLecture().getCourse())) {
```

Comment:
Keep the indentions low:
```suggestion
        if (!lectureIngestionEnabled(attachmentUnits.getFirst().getLecture().getCourse())) {
            return false;
        }
        try {
```